library/clp/simplex.pl

```diff
@@ -1044,15 +1044,15 @@
 
 row_eliminate(row(Var, Ls0, R0), row(_, PLs, PR), Col, row(Var, Ls, R)) :-
         nth0(Col, Ls0, Coeff),
-        (   Coeff =:= 0 -> Ls = Ls0, R = R0
-        ;   MCoeff is -Coeff,
-            all_times_plus([PR|PLs], MCoeff, [R0|Ls0], [R|Ls])
+        (   Coeff = 0 -> Ls = Ls0, R = R0  % assumes rational Coeff
```

Comment:
Do you have any proof this helps?  I tried the code below using `-O` (or setting the Prolog flag `optimise` to `true` before compiling).   The timing for 10 000 000 iterations are 0.192, 0.201 and 0.153.  That is more or less in line with what I expected.   =/2 is faster than =:=/2, but as a condition inside if-then-else Prolog can turn =:=/2 into a fast switch because no variables are instantiated.  Using ==/2 is best as it both uses the simplified branch and the test is simpler.

```
l0(N) :-
    (   N =:= 0
    ->  true
    ;   N2 is N - 1,
        l0(N2)
    ).

l1(N) :-
    (   N = 0
    ->  true
    ;   N2 is N - 1,
        l1(N2)
    ).

l2(N) :-
    (   N == 0
    ->  true
    ;   N2 is N - 1,
        l2(N2)
    ).
```